src/app/App.js

```diff
@@ -924,7 +940,7 @@ class App extends EventEmitter {
 			historyState.scrollTop = this.popstateScrollTop;
 			historyState.scrollLeft = this.popstateScrollLeft;
 		}
-		this.updateHistory_(title, redirectPath, nextScreen.beforeUpdateHistoryState(historyState), opt_replaceHistory);
```

Comment:
I've also added a test case to make sure the history state is always updated with the redirectPath so we don't let this slide again.